lua/autorun/server/sv_pvp_movespeed.lua

```diff
@@ -64,23 +67,39 @@ local function adjustMovementSpeed(ply)
     local wepCount = 0
     
     -- count weapons
-    for k, weapon in pairs(weapons) do
+    for _, weapon in pairs( weapons ) do
         if nonEffectedWeapons[weapon:GetClass()] == nil then
-            wepCount =  wepCount+1
+            wepCount =  wepCount + 1
         end
     end
+    wepCount = math.max(wepCount + wepNum, 0)
     
-    local multiplier = movementMultiplier(wepCount) 
+    local multiplier = movementMultiplier( wepCount ) 
     setSpeed(ply, multiplier)
 end
 
 -- Hook Functions --
-local function onEquipped( wep, ply )
+local function onEquip( wep, ply )
     if not IsValid( ply ) then return end
-    adjustMovementSpeed( ply )
+    local wepNum = 0
```

Comment:
This part feels weird to me. Is `wepNum` going to only ever be 0 or 1?

Reply:
ye it is weird, and could probably use a better solution
https://github.com/CFC-Servers/cfc_pvp_movespeed/pull/3#discussion_r222516352

Reply:
Try it out, make a gist and we can talk about it then @PLally